app/static/js/index.js

```diff
@@ -2,380 +2,607 @@
+			filter.classList.toggle("active", !checked);
+			for (let checkbox of otherCheckboxes) {
+				checkbox.checked = !checked;
+			}
+		} else {
+			if (checked) {
+				filter.classList.add("active");
+				allCheckbox.checked = false;
+			} else {
+				let anyChecked = false;
+				for (let checkbox of otherCheckboxes) {
+					if (checkbox.checked) {
+						anyChecked = true;
+						break;
+					}
+				}
+				filter.classList.toggle("active", anyChecked);
+				allCheckbox.checked = !anyChecked;
+			}
+		}
+		runSearch();
+	}
 };
 
+p.scrollTop.onclick = function () {
+	window.scrollTo({
+		top: 0,
+		behavior: "smooth",
+	});
+};
 
 ///////////////////
 // List building //
 ///////////////////
 let criteria = {
-    'filters': {},
+	filters: {},
 };
 let pagesLoaded = 0;
 let pagesFinished = false;
 
 function runSearch() {
-    let filters = {};
-    for (let filter of p.filters) {
-        let category = filter.id;
-        let otherCheckboxes = Array.from(filter.getElementsByTagName('input'));
-        let allCheckbox = otherCheckboxes.shift();
-        if (!allCheckbox.checked) {
-            filters[category] = [];
-            for (let checkbox of otherCheckboxes) {
-                if (checkbox.checked) {
-                    if (category === 'leave') {
-                        if (checkbox.name === 'True') {
-                            filters[category].push(true);
-                        } else if (checkbox.name === 'False') {
-                            filters[category].push(false);
-                        } else {
-                            filters[category].push(checkbox.name || null);
-                        }
-                    } else if (['year', 'birth_month', 'birth_day', 'floor', 'room'].includes(category)) {
-                        filters[category].push(checkbox.name ? parseInt(checkbox.name) : null);
-                    } else {
-                        filters[category].push(checkbox.name);
-                    }
-                }
-            }
-        }
-    }
-    criteria = {};
-    query = p.query.value.trim();
-    if (query)
-        criteria['query'] = query;
-    criteria['filters'] = filters;
-    p.list.innerHTML = '';
-    pagesLoaded = 0;
-    pagesFinished = false;
-    loadNextPage();
+	let filters = {};
+	for (let filter of p.filters) {
+		let category = filter.id;
+		let otherCheckboxes = Array.from(filter.getElementsByTagName("input"));
+		let allCheckbox = otherCheckboxes.shift();
+		if (!allCheckbox.checked) {
+			filters[category] = [];
+			for (let checkbox of otherCheckboxes) {
+				if (checkbox.checked) {
+					if (category === "leave") {
+						if (checkbox.name === "True") {
+							filters[category].push(true);
+						} else if (checkbox.name === "False") {
+							filters[category].push(false);
+						} else {
+							filters[category].push(checkbox.name || null);
+						}
+					} else if (
+						["year", "birth_month", "birth_day", "floor", "room"].includes(
```

Comment:
Yeah this beautification is making the code actively harder to read... I would prefer it if we didn't do this

Reply:
I think the code is pretty clean already